File: src/website/services/notifications/subscriptionExists.ts
import CanisterClientFactory from "../CanisterClientFactory";
import { toCandid as userIdToCandid } from "../candidConverters/userId";
import { UserId } from "../../domain/model/users";
import { SubscriptionExistsArgs } from "./candid/types";

export default async function(userId: UserId, p256dh_key: string) : Promise<boolean> {
    const client = CanisterClientFactory.current!.notificationsClient;
    const args: SubscriptionExistsArgs = {
        user_id: userIdToCandid(userId),
        p256dh_key
    };    
    const response = await client.subscription_exists(args);
    return "Yes" in response;
}

File: v2/backend/canisters/notifications/api/src/queries/subscription_exists.rs
use candid::CandidType;
use serde::Deserialize;
use types::UserId;

#[derive(CandidType, Deserialize, Debug)]
pub struct Args {
    pub user_id: UserId,
    pub p256dh_key: String,
}

#[derive(CandidType, Deserialize, Debug)]
pub enum Response {
    Yes,
    No,
}
